src/main/java/mr/springbootapi/service/MenuItemServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package mr.springbootapi.service;
+
+import mr.springbootapi.entity.MenuItem;
+import mr.springbootapi.exception.InvalidPriceException;
+import mr.springbootapi.exception.ItemNotFoundException;
+import mr.springbootapi.repository.MenuItemRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MenuItemServiceImpl implements MenuItemService {
+    private final MenuItemRepository menuItemRepository;
+
+    public MenuItemServiceImpl(MenuItemRepository menuItemRepository) {
+        this.menuItemRepository = menuItemRepository;
+    }
+
+    @Override
+    public MenuItem addMenuItem(MenuItem menuItem) {
+        if (menuItem.getPrice() < 0) {
+            throw new InvalidPriceException("Invalid price");
+        }
+        return menuItemRepository.save(menuItem);
+    }
+
+    @Override
+    public List<MenuItem> listAllMenuItems() {
+        return menuItemRepository.findAll();
+    }
+
+    @Override
+    public MenuItem getById(int id) {
+        return menuItemRepository
+                .findById(id)
+                .orElseThrow(() -> new ItemNotFoundException("Item not found"));
```

Comment:
method chaining is the way to go, but since the code is going to be read by beginners it's better to keep the if/else so everything is clear,
```suggestion
    /*
     Use the repository to find a menu item with the given ID.
     The result is wrapped in an Optional object to handle the possibility
     of not finding a matching item.
    */
    Optional<MenuItem> menuItemOptional = menuItemRepository.findById(id);
    // Check if the Optional object contains a value (meaning a menu item was found)
    if (menuItemOptional.isPresent()) {
      // If a menu item is found, extract it from the Optional object and return it
      return menuItemOptional.get();
    } else {
      // If no menu item is found, throw an ItemNotFoundException
      throw new ItemNotFoundException("Item not found") ;
    }
                
```